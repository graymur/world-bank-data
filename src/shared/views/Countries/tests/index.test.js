/* eslint-disable */
import React from 'react';
import renderer from 'react-test-renderer';
import {StaticRouter} from 'react-router';
import {Countries} from '../';
import _ from 'lodash';

const props = {
	data: {
		loading: false,
		countries: [{
			iso2Code: 'AA',
			name: 'First name',
		}, {
			iso2Code: 'BB',
			name: 'Second name',
		}]
	}
};

test('Renders countries list', () => {
	const component = renderer.create(
		<StaticRouter location="someLocation" context={{}}>
			<Countries {...props}/>
		</StaticRouter>
	);

	let tree = component.toJSON();
	expect(tree).toMatchSnapshot();
});
