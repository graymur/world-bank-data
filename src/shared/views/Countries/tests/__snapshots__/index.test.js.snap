// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Calls loadCountries prop if countries array is empty 1`] = `
<div
  className="countries"
>
  <h1
    className="countries__title"
  >
    Countries List
  </h1>
  <div
    className="countries__container"
  >
    <nav
      className="countries__list"
    />
  </div>
</div>
`;

exports[`Renders countries list 1`] = `
<div
  className="countries"
>
  <h1
    className="countries__title"
  >
    Countries List
  </h1>
  <div
    className="countries__container"
  >
    <nav
      className="countries__list"
    >
      <a
        className="countries__list__item"
        href="/countries/AA"
        onClick={[Function]}
      >
        First name
      </a>
      <a
        className="countries__list__item"
        href="/countries/BB"
        onClick={[Function]}
      >
        Second name
      </a>
    </nav>
  </div>
</div>
`;
